hack/version.py

```diff
@@ -6,167 +6,154 @@
-        or that's not 'ocm-X.Y' or (a descendant of) master.
+        or that's not 'mce-X.Y' or (a descendant of) master.
     """
-    # We're cloning the hive repo, so there's no local branch for ocm-X.Y[-mce-M.N]; but we don't want to
-    # force the user to say 'origin/ocm-X.Y[-mce-M.N]'. Accommodate...
-    parsed, remote, _, _ = parse_branch_name(branch_arg)
+    # We're cloning the hive repo, so there's no local branch for mce-X.Y; but we don't want to
+    # force the user to say 'origin/mce-X.Y'. Accommodate...
+    parsed, remote, _ = parse_mce_branch_name(branch_arg)
     if parsed and not remote:
         branch_arg = "origin/{}".format(branch_arg)
 
     # This will raise an exception if there's no such commit-ish
-    commit_hash = hive_repo.rev_parse(branch_arg).hexsha
-
-    # We need to know where master is, even if we're processing a different branch. However, if
-    # we've cloned from something other than HIVE_REPO_DEFAULT, there may not be a local `master`
-    # branch. So we may need to try origin and upstream to find it.
-    for remote in ("", "origin/", "upstream/"):
-        master_branch = remote + HIVE_BRANCH_DEFAULT
-        log("Trying to find master at {}".format(master_branch))
-        try:
-            master_hash = hive_repo.rev_parse(master_branch).hexsha
-        except git.BadName:
-            log("Couldn't find master at `{}`".format(master_branch))
-        else:
-            log("Found master at `{}`".format(master_branch))
-            break
-    else:
-        log("Couldn't find master branch!")
-        sys.exit(1)
+    commit = hive_repo.rev_parse(branch_arg)
+    master_commit = find_branch(hive_repo, HIVE_BRANCH_DEFAULT)
 
-    is_master_ancestor = is_ancestor(hive_repo, commit_hash, master_hash)
+    # This includes when the commit is equal to the master commit
+    is_master_ancestor = hive_repo.is_ancestor(commit, master_commit)
 
-    # Was (a commit on) an ocm-X.Y[-mce-M.N] branch specified?
-    # This will find all such branch names and add them to the channel list.
-    # However, we'll only support >1 entry in this list if we're tracking
-    # master, because otherwise which would we use to compute the semver?
-    channels = []
+    # Was (a commit on) an mce-X.Y branch specified?
+    # This will find all such branch names containing this commit
     maybe_prefix = None
-    for info in remote_branch_info(hive_repo).values():
-        if info["commit_hash"] == commit_hash:
-            channels.append(info["channel"])
+    for info in remote_mce_branch_info(hive_repo).values():
+        if info["commit_hash"] == commit.hexsha:
             # We'll only use this if we end up with one
             # entry in this list AND we're not tracking master
             maybe_prefix = info["prefix"]
 
-    if commit_hash == master_hash or (is_master_ancestor and not maybe_prefix):
+    if is_master_ancestor and not maybe_prefix:
```

Comment:
Not sure if there is a case where `commit_hash == master_hash and maybe_prefix`?

Reply:
I think we have the opportunity to simplify the code so that question becomes moot.

All we want to get out of this function is a commit sha (easy) and a prefix. To get the prefix we just want to know whether we're dealing with `master` or an `mce-*` branch. If the latter, we parsed out the prefix; use it. If the former, use the default const.

We may be able to rewrite `remote_mce_branch_info` to be more generally `branch_info`, containing entries for `master` as well as `mce-*`....